packages/shylu/shylu_node/hts/src/shylu_hts_impl_def.hpp

```diff
@@ -4044,41 +4045,47 @@ ondiag_solve (const OnDiagTri& t, Sclr* x, const Int ldx, const Int nrhs,
   const Int nthreads = t.nthreads();
   if (nthreads == 1) {
     t.solve(x, ldx, x, ldx, nrhs);
+#ifdef _OPENMP
+#   pragma omp flush
+#endif
     *t_barrier = step;
   } else {
     // Solve T wrk_ = x.
     t.solve(x, ldx, wrk_.data(), t.get_n(), nrhs);
-    { // Wait for the block row MVPs to finish.
```

Comment:
Why was this moved down?

Reply:
Sync of memory has to occur before the lock acquistion.

Reply:
Ah, to ensure that the store is written to global memory for other threads to read.